tests/unit/test_file_handle_cache.py

```diff
@@ -0,0 +1,52 @@
+"""
+Tests the file handle cache
+"""
+
+from ga4gh import datamodel
+
+import unittest
+import tempfile
+import os
+
+import ga4gh.frontend as frontend
+
+
+class TestFileHandleCache(datamodel.PysamFileHandleCache, unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super(TestFileHandleCache, self).__init__()
+        unittest.TestCase.__init__(self, *args, **kwargs)
+        frontend.configure(baseConfig="TestConfig")
+
+    def _getFileHandle(self, dataFile):
+        def openMethod(dataFile):
+            return open(dataFile, 'w')
+        return self.getFileHandle(dataFile, openMethod)
+
+    def testGetFileHandle(self):
+        def removeFiles(l):
+            for f in l:
+                os.remove(f)
+
+        def genFileName(x):
+            return os.path.join(tempfile.gettempdir(),
```

Comment:
Any reason you can't open a second handle to this via the file name? It might be a bit tricky if you're doing IO, but I think it'll probably work if you're just opening the file.